frontend/controller/service-worker.js

```diff
@@ -51,6 +51,34 @@ sbp('sbp/selectors/register', {
       await swRegistration.update()
       setInterval(() => sbp('service-worker/update'), HOURS_MILLIS)
 
+      // Send a 'ready' message to the SW and wait back for a response
```

Comment:
There's another issue in this file, this `'service-workers/setup'` doesn't throw if it fails, it catches the error and logs it only:

```js
    } catch (e) {
      console.error('error setting up service worker:', e)
    }
```

This means this code in `main.js` won't get triggered:

```js
  // register service-worker
  await Promise.race(
    [sbp('service-workers/setup'),
      new Promise((resolve, reject) => {
        setTimeout(() => {
          reject(new Error('Timed out setting up service worker'))
        }, 16e3)
      })]
  ).catch(e => {
    console.error('[main] Error setting up service worker', e)
    alert(L('Error while setting up service worker'))
    window.location.reload() // try again, sometimes it fixes it
    throw e
  })
```

Reply:
I've added a throw